### PR TITLE
chore: use vitest workspaces

### PR DESCRIPTION
**Pull request by juliusmarminge**
Closes https://github.com/trpc/trpc/pull/5262

## 🎯 Changes

What changes are made in this PR? Is it a feature or a bug fix?

## ✅ Checklist

- [ ] I have followed the steps listed in the [Contributing guide](https://github.com/trpc/trpc/blob/main/CONTRIBUTING.md).
- [ ] If necessary, I have added documentation related to the changes made.
- [ ] I have added or updated the tests related to the changes made.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -39,13 +39,14 @@ jobs:
...
-        run: pnpm turbo --filter tests test-ci
+      - name: Build tRPC, then test
+        run: pnpm turbo --filter "@trpc/*" build
+      - run: pnpm vitest --coverage
 
-      - uses: codecov/codecov-action@v4
-        with:
-          # token: ${{ secrets.CODECOV_TOKEN }} # not required for public repos
-          directory: ./packages
+      - uses: codecov/codecov-action@v3
```
Comment (juliusmarminge) [inline .github/workflows/main.yml:46]: REVERT ME

v4 requires a token: 
image

Comment (KATT) [reply .github/workflows/main.yml:46]: We have a `CODECOV_TOKEN` in the repo already that should work

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -39,13 +39,14 @@ jobs:
...
+        run: pnpm turbo --filter "@trpc/*" build
+      - run: pnpm test-ci
 
-      - uses: codecov/codecov-action@v4
-        with:
-          # token: ${{ secrets.CODECOV_TOKEN }} # not required for public repos
-          directory: ./packages
+      - uses: codecov/codecov-action@v3
+        # with:
+        #   # token: ${{ secrets.CODECOV_TOKEN }} # not required for public repos
```
Comment (KATT) [inline .github/workflows/main.yml:48]: this token should work

File: vitest.config.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    environment: 'jsdom',
+    globals: true,
+    snapshotFormat: {
+      printBasicPrototype: true,
+    },
+    setupFiles: ['./tests/setupTests.ts'],
+    coverage: {
+      provider: 'istanbul',
+      include: ['**/src/**'],
+      exclude: ['**/www/**', '**/examples/**'],
```
Comment (KATT) [inline vitest.config.ts:48]: maybe simpler like 
```suggestion
      include: ['packages/**/*'],
      exclude: ['packages/tests'],
```

**Review by KATT — APPROVED** [review]

File: vitest.config.ts (hunk 3 above)
Comment (juliusmarminge) [reply vitest.config.ts:48]: i never found out what this pattern is relative to, so this was the only way i got it reliably working...

**Pull request merged**